engine/security/userns-remap.md

```diff
@@ -0,0 +1,249 @@
+---
+description: Isolate containers within a user namespace
+keywords: security, namespaces
+title: Isolate containers with a user namespace
+---
+
+Linux namespaces provide isolation for running processes, limiting their access
+to system resources without the running process being aware of the limitations.
+For more information on Linux namespaces, see
+[Linux namespaces](https://www.linux.com/news/understanding-and-securing-linux-namespaces){: target="_blank" class="_" };
+
+The best way to prevent privilege-escalation attacks from within a container is
+to configure your container's applications to run as unprivileged users. For
+containers whose processes must run as the `root` user within the container, you
+can re-map this user to a less-privileged user on the Docker host. The mapped
+user is assigned a range of UIDs which function within the namespace as normal
+UIDs from 0 to 65536, but have no privileges on the host machine itself.
+
+## About remapping and subordinate user and group IDs
+
+The remapping itself is handled by two files: `/etc/subuid` and `/etc/subgid`.
+Each file works the same, but one is concerned with the user ID range, and the
+other with the group ID range. Consider the following entry in `/etc/subuid`:
+
+```none
+testuser:231072:65536
+```
+
+This means that `testuser` is assigned a subordinate user ID range of `230172`
+and the next 65536 integers in sequence. UID `231072` is mapped within the
+namespace (within the container, in this case) as UID `0` (`root`). UID `231073`
+is mapped as UID `1`, and so forth. If a process attempts to escalate privilege
+outside of the namespace, the process is running as an unprivileged high-number
+UID on the host, which does not even map to a real user. This means the process
+has no privileges on the host system at all.
+
+> Multiple ranges
```

Comment:
should we have an example here?  It's not clear what happens if you have multiple ranges, what the expected behavior should be, or if there is any other user action required to make use of them.